cmd/publish-failure-resolver-go/main.go

```diff
@@ -129,6 +144,53 @@ func main() {
 	}
 }
 
+// Filters out entries originating from Methode OR Wordpress from a list of UUIDs.
+// Used to prevent republishing of entries which are due update.
+func filterAllowListedUUIDs(uuids []string, filterListPath string) ([]string, error) {
+	file, err := os.Open(filterListPath)
+	if err != nil {
+		return nil, fmt.Errorf("opening file: %w", err)
+	}
+	defer file.Close()
+
+	b, err := ioutil.ReadAll(file)
+	if err != nil {
+		return nil, fmt.Errorf("reading file: %w", err)
+	}
+
+	var filterList map[string][]string
+
+	if err = json.Unmarshal(b, &filterList); err != nil {
+		return nil, fmt.Errorf("unmarshalling records: %w", err)
+	}
+
+	contentProvider := "universal-content"
+	res := make([]string, 0, len(uuids))
+
+	for _, uuid := range uuids {
+		var allowListed bool
```

Comment:
To simplify the if/else I would recommend

```
		var allowListed bool

		providers, ok := filterList[uuid]
		allowListed = ! ok
		if ok {
		    allowListed = ElementExists(contentProvider, providers) //where ElementExists is helper function that returns bool
		}

		if allowListed {
			res = append(res, uuid)
		}
```

Reply:
Yes, I agree it could be simplified however I'd prefer not to introduce such a small function if possible.
May I suggest the following or do you prefer the helper func instead?

```
providers, ok := filterList[uuid]
if !ok {
	res = append(res, uuid)
	continue
}

for _, p := range providers {
	if p == contentProvider {
		// Entries originating from "universal-content" are safe to republish.
		res = append(res, uuid)
		break
	}
}
```

Reply:
The above is just a recommendation so it is up to you how to proceed :)

One of the main "rules" I follow when creating a function is that the function should do one thing and do it well. So it looks like that searching for an element in an array is another task that can be extracted and implemented separately. As a side bonus the code can be re-written to 4 lines.